API/APIManager.m

```diff
@@ -111,30 +109,32 @@ - (void)fetchNews: (void(^)(NSArray *newsArticles, NSError *error))completion{
 
 - (void)fetchHeadlineNews:(NSString *)ticker completion:(void(^)(NSMutableArray *allNewsArticles, NSError *error))completion {
     NSString *keyword = ticker;
-    NSString *appendEndpoint = [headlinesApiUrl stringByAppendingString:keyword];
-    NSURL *urlforSpecificHeadlines = [NSURL URLWithString:appendEndpoint];
-    NSMutableURLRequest *requestForHeadlines = [NSMutableURLRequest requestWithURL:urlforSpecificHeadlines cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
-    [requestForHeadlines setHTTPMethod:@"GET"];
-    [requestForHeadlines setValue:@"application/json" forHTTPHeaderField: @"Content-Type"];
-    NSString *path = [[NSBundle mainBundle] pathForResource: @"Keys" ofType: @"plist"];
-    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
-    NSString *key = [dict objectForKey: @"NewsArticles"];
-    [requestForHeadlines addValue:key forHTTPHeaderField:@"X-Api-Key"];
-    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration] delegate:nil delegateQueue:[NSOperationQueue mainQueue]];
-    NSURLSessionDataTask *task = [session dataTaskWithRequest:requestForHeadlines completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
-           if (error != nil) {
-               NSLog(@": YD: %@", [error localizedDescription]);
-           }
-          else {
-              NSDictionary *newsDictionary = [NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingMutableContainers error:nil];
-              NSMutableArray *storylinesForKeyword = newsDictionary[@"articles"];
-              NSMutableArray *keywordArticleOfNews = [News arrayOfNews:storylinesForKeyword];
-              completion(keywordArticleOfNews, nil);
-
-          }
-    }];
-}
+    if(ticker != nil){
+        NSString *appendEndpoint = [headlinesApiUrl stringByAppendingString:keyword];
+        NSURL *urlforSpecificHeadlines = [NSURL URLWithString:appendEndpoint];
+        NSMutableURLRequest *requestForHeadlines = [NSMutableURLRequest requestWithURL:urlforSpecificHeadlines cachePolicy:NSURLRequestReloadIgnoringLocalCacheData timeoutInterval:10.0];
```

Comment:
You could declare `timeoutInterval` as a constant because you using the same value of `10.0` in multiple places